vendors/st/boards/stm32l475_discovery/ports/common_io/iot_uart.c

```diff
@@ -472,16 +478,71 @@ int32_t iot_uart_ioctl( IotUARTHandle_t const pxUartPeripheral,
                 break;
 
             case eUartGetConfig:
-
-                ( ( IotUARTConfig_t * ) pvBuffer )->ulBaudrate = pxUartPeripheral->pxHuart->Init.BaudRate;
-                ( ( IotUARTConfig_t * ) pvBuffer )->xParity = pxUartPeripheral->pxHuart->Init.Parity;
-                ( ( IotUARTConfig_t * ) pvBuffer )->ucWordlength = pxUartPeripheral->pxHuart->Init.WordLength;
-                ( ( IotUARTConfig_t * ) pvBuffer )->xStopbits = pxUartPeripheral->pxHuart->Init.StopBits;
-                ( ( IotUARTConfig_t * ) pvBuffer )->ucFlowControl = pxUartPeripheral->pxHuart->Init.HwFlowCtl;
-
-                lError = IOT_UART_SUCCESS;
-
-                break;
+               {
+                   IotUARTConfig_t * pConfigBuffer = ( IotUARTConfig_t * ) pvBuffer;
+
+                   pConfigBuffer->ulBaudrate = pxUartPeripheral->pxHuart->Init.BaudRate;
+
+                   switch( pxUartPeripheral->pxHuart->Init.Parity )
+                   {
+                       case UART_PARITY_NONE:
```

Comment:
Where are these macro come from? I do not see them in the internal branch.

Reply:
In the file path, you can see this is specific implementation on ST board. Therefore the macro is part of ST board's driver.

I didn't add ST implementation to internal branch.